util.py

```diff
@@ -279,3 +279,17 @@ def check_file_sha256_checksum(filename, reference_checksum):
     :param str filename: a single file to be checked
     """
     assert compute_file_sha256_checksum(filename) == reference_checksum
+
+
+def resolve_sis_paths_in_dict(dictionary):
```

Comment:
sisyphus has a function `extract_paths` which does a very similar thing ([here](https://github.com/rwth-i6/sisyphus/blob/master/sisyphus/tools.py#L74)). Maybe that could be used here as well? On the other hand, this here is probably easier to read.

Reply:
I know about this function. The problem is, that it flattens the data structure in the process (I.e. Returned is a flat list of all path-strings).
I didn't find an elegant solution to write it back in the dict structure.

Reply:
Alright, then let's keep it as it is.